### PR TITLE
langchain4j-chatglm get rid of lombok

### LINKED ISSUE #1636: [FEATURE] Get rid of Lombok
Rewrite lombok-generated code and get rid of lombok dependency.

### PR DESCRIPTION
**Pull request by Martin7-1**
## Issue
Closes #1636 

## Change
1. `langchain4j-chatglm` get rid of lombok
2. added `logRequests` and `logResponses` parameters

## General checklist
- [x] There are no breaking changes
- [ ] I have added unit and integration tests for my change
- [x] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [x] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green
- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

### CODE REVIEW

**Comment by langchain4j** [comment]
@Martin7-1 there are conflicts, could you please check? Seems Ollama changes somehow got into this branch

**Comment by Martin7-1** [comment]
> @Martin7-1 there are conflicts, could you please check? Seems Ollama changes somehow got into this branch

Done.

File: langchain4j-chatglm/src/main/java/dev/langchain4j/model/chatglm/ChatGlmResponseLoggingInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+public class ChatGlmResponseLoggingInterceptor implements Interceptor {
```
Comment (langchain4j) [inline langchain4j-chatglm/src/main/java/dev/langchain4j/model/chatglm/ChatGlmResponseLoggingInterceptor.java]:
```suggestion
class ChatGlmResponseLoggingInterceptor implements Interceptor {
```

File: langchain4j-chatglm/src/main/java/dev/langchain4j/model/chatglm/ChatGlmRequestLoggingInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.StreamSupport;
+
+import static dev.langchain4j.internal.Utils.isNullOrBlank;
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.joining;
+
+public class ChatGlmRequestLoggingInterceptor implements Interceptor {
```
Comment (langchain4j) [inline langchain4j-chatglm/src/main/java/dev/langchain4j/model/chatglm/ChatGlmRequestLoggingInterceptor.java]:
```suggestion
class ChatGlmRequestLoggingInterceptor implements Interceptor {
```

**Review by ghost — DISMISSED** [review]
@Martin7-1 thank you!

**Review by ghost — APPROVED** [review]
@Martin7-1 thank you!

**Pull request merged**